hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/WriteBufferedP.java

```diff
@@ -52,7 +53,11 @@
 
     @Override
     public void process(int ordinal, @Nonnull Inbox inbox) {
-        inbox.drain((T t) -> addToBuffer.accept(buffer, t));
+        for (Object o; (o = inbox.poll()) != null; ) {
+            if (!(o instanceof Punctuation)) {
+                addToBuffer.accept(buffer, (T) o);
+            }
+        }
```

Comment:
Not so much simpler: 

        inbox.drain(item -> {
            if (!(item instanceof Punctuation)) {
                addToBuffer.accept(buffer, (T) item);
            }
        });

And with plain old java it's easier to read.